sklearn/utils/multiclass.py

```diff
@@ -266,26 +267,43 @@ def type_of_target(y):
         pass
 
     # Invalid inputs
-    if y.ndim > 2 or (y.dtype == object and len(y) and
-                      not isinstance(y.flat[0], str)):
-        return 'unknown'  # [[[1, 2]]] or [obj_1] and not ["label_1"]
-
-    if y.ndim == 2 and y.shape[1] == 0:
+    if y.ndim not in (1, 2):  # [[[1, 2]]]
+        return 'unknown'
+    if not min(y.shape):
+        if y.ndim == 1:
+            return 'binary'  # []
         return 'unknown'  # [[]]
+    # [obj_1] and not ["label_1"]
+    if not issparse(y) and y.dtype == object and \
+            not isinstance(y.flat[0], str):
+        return 'unknown'
+    if issparse(y) and y.dtype == object and not isinstance(y.data[0], str):
```

Comment:
Why are we handling the sparse + str case? This does not appear to be tested. What does it mean?

Reply:
Sorry for the delay, no idea why I did that at the time 🤔 anyway I removed it, indeed it was not covered by any test